server/sonar-server/src/main/java/org/sonar/server/branch/pr/ws/ListAction.java

```diff
@@ -110,15 +111,16 @@ public void handle(Request request, Response response) throws Exception {
   private static void addPullRequest(ProjectPullRequests.ListWsResponse.Builder response, BranchDto branch, Map<String, BranchDto> mergeBranchesByUuid,
     BranchStatistics branchStatistics, @Nullable String analysisDate) {
     Optional<BranchDto> mergeBranch = Optional.ofNullable(mergeBranchesByUuid.get(branch.getMergeBranchUuid()));
+
     ProjectPullRequests.PullRequest.Builder builder = ProjectPullRequests.PullRequest.newBuilder();
-    String branchKey = branch.getKey();
-    builder.setId(branchKey);
-    //TODO get the branch from PULL_REQUEST_DATA
-    builder.setBranch(branchKey);
-    //TODO get the URL from PULL_REQUEST_DATA
-    builder.setUrl("https://github.com/SonarSource/sonar-core-plugins/pull/32");
-    //TODO get the branch title from PULL_REQUEST_DATA
-    builder.setTitle(branchKey);
+    builder.setId(branch.getKey());
+
+    // known to be non-null; all pull requests have non-null value, and caller filters accordingly
+    DbProjectBranches.PullRequestData pullRequestData = branch.getPullRequestData();
+    builder.setBranch(pullRequestData.getBranch());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "pullRequestData" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fuse-pr-id-as-key-for-pull-requests&issues=AWG38FNK42pHCPIO0Eiw&open=AWG38FNK42pHCPIO0Eiw)

[ISSUE_KEY]: AWG38FNK42pHCPIO0Eiw